crates/core/protobuf/node_ctl_svc.proto

```diff
@@ -20,6 +21,29 @@ service NodeCtlSvc {
   rpc GetIdent(google.protobuf.Empty) returns (IdentResponse);
 
   rpc GetMetadata(GetMetadataRequest) returns (GetMetadataResponse);
+
+  // Provision the Restate cluster on this node.
+  rpc ProvisionCluster(ProvisionClusterRequest) returns (ProvisionClusterResponse);
+}
+
+message ProvisionClusterRequest {
+  bool dry_run = 1;
+  optional uint32 num_partitions = 2;
+  optional restate.cluster.ReplicationStrategy placement_strategy = 3;
+  optional restate.cluster.DefaultProvider log_provider = 4;
+}
+
+enum ProvisionClusterResponseKind {
+  ProvisionClusterResponseType_UNKNOWN = 0;
+  DRY_RUN = 1;
+  NEWLY_PROVISIONED = 2;
+  ALREADY_PROVISIONED = 3;
+}
+
+message ProvisionClusterResponse {
+  ProvisionClusterResponseKind kind = 1;
+  // This field will be empty if the cluster is already provisioned
+  optional restate.cluster.ClusterConfiguration cluster_configuration = 3;
```

Comment:
The main reason why I didn't return it was that it would have required syncing the metadata to obtain the latest metadata from which to reconstruct the `ClusterConfiguration`.